pkgs/games/ldmud/default.nix

```diff
@@ -0,0 +1,107 @@
+{ lib, fetchFromGitHub, stdenv, autoreconfHook, pkg-config, bison, libiconv
+, pcreSupport ? true, pcre, gcryptSupport ? true, libgcrypt, erqSupport ? "erq"
+, compatModeSupport ? false, filenameSpacesSupport ? false, ipv6Support ? false
+, mccpSupport ? false, zlib, mysqlSupport ? false, libmysqlclient
+, postgresSupport ? false, postgresql, sqliteSupport ? false, sqlite
+, jsonSupport ? false, json_c, xmlSupport ? false, libxml2
+, deprecatedSupport ? true, tlsSupport ? false, openssl, pythonSupport ? false
+, python39, pythonSupportVersion ? python39 }:
+
+let supportedErq = [ "no" "erq" "xerq" ];
+in assert builtins.elem erqSupport supportedErq;
+
+stdenv.mkDerivation rec {
+  pname = "ldmud";
+  version = "3.6.6";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-2TaFt+T9B5Df6KWRQcbhY1E1D6NISb0oqLgyX47f5lI=";
+    name = pname;
+  };
+
+  # LD keeps its source code in a 'src' subfolder. Note the 'sourceRoot' is
+  # relative to the tmp build directory nix creates so we need the predictable
+  # 'name' set above.
+  sourceRoot = "${pname}/src";
+
+  # Autoreconf, pkg-config and bison are required in all cases.
+  # Other dependencies are optional.
+  nativeBuildInputs = [ autoreconfHook pkg-config bison ]
+    ++ lib.optional pcreSupport pcre ++ lib.optional gcryptSupport libgcrypt
+    ++ lib.optional mccpSupport zlib ++ lib.optional mysqlSupport libmysqlclient
+    ++ lib.optional postgresSupport postgresql
+    ++ lib.optional sqliteSupport sqlite ++ lib.optional jsonSupport json_c
+    # NB: LD also supports iksemel for the XML backend, but libxml2 is much
+    # better supported so we don't offer a choice here.
+    ++ lib.optional xmlSupport libxml2
+    # NB: LD also supports gnutls as a backend. I'm unconvinced we need a toggle
+    # here and so default to openssl.
+    ++ lib.optional tlsSupport openssl
+    ++ lib.optional pythonSupport pythonSupportVersion;
+  # On Darwin we need to include libiconv explicitly or there will be build errors
+  # about missing symbols.
+  buildInputs = lib.optionals stdenv.isDarwin [ libiconv ];
+
+  # To support systems without autoconf LD puts its `configure.ac` in a subdir
+  # and uses a small wrapper script, `src/autogen.sh` to process it. This script
+  # checks if autoconf is installed and either runs it and copies the generated
+  # files to the src root, or it copies pre-generated vendored copies to the src
+  # root.
+  #
+  # We always have autoconf and a nice stdenv autoreconfHook we can use, but only
+  # when configure.ac is in the expected place in the srcroot. Symlink to the
+  # rescue.
```

Comment:
Can we short this comment to one or two lines?

Reply:
Definitely.